service-base/src/main/java/org/eclipse/hono/servicebase/HonoServiceBase.java

```diff
@@ -146,15 +159,25 @@ public String getBindAddress() {
      *
      * @return The IP address.
      */
-    public String getInsecurePortBindAddress() {
+    public final String getInsecurePortBindAddress() {
         return config.getInsecurePortBindAddress();
     }
 
-    public static boolean isPortUnconfigured(int portToCheck) {
+    /**
+     * Checks if the given port number represents an unconfigured port.
+     * @param portToCheck The port number to check.
+     * @return True if the port number if unconfigured.
+     */
+    public static final boolean isPortUnconfigured(int portToCheck) {
         return (portToCheck == Constants.PORT_UNCONFIGURED);
     }
 
-    public static boolean isPortConfigured(int portToCheck) {
+    /**
+     * Checks if the given port number represents an explicitly configured port.
+     * @param portToCheck The port number to check.
+     * @return True if the port number was configured explicitly.
+     */
+    public static final boolean isPortConfigured(int portToCheck) {
```

Comment:
why isn't a port number of `0` considered *configured*? The user explicitly configured the port to 0 so that a free port is picked during start-up ...